aten/src/ATen/core/TensorMethods.h

```diff
@@ -1800,7 +1800,8 @@ inline bool is_quantized(Tensor self) {
     return static_cast<T*>(this->data_ptr());    \
   }
 
-AT_FORALL_SCALAR_TYPES_WITH_COMPLEX_EXCEPT_COMPLEX_HALF(DEFINE_CAST)
+AT_FORALL_SCALAR_TYPES_WITH_COMPLEX_EXCEPT_COMPLEX_HALF_AND_QINT(DEFINE_CAST)
+AT_FORALL_QINTS(DEFINE_CAST)
```

Comment:
do we want to cast them to quint8* or uint8*?

Reply:
I feel `uint8*` might be more useful for people

Reply:
ya, I think that's the correct solution but outside of the scope of this PR, which is just trying to rationalize the AT_FORALL macros.  So I'm just trying to match up with this one: https://github.com/pytorch/pytorch/blob/34f53564b4e5937780a9784e7694c4df98c217cb/aten/src/ATen/Dispatch.h#L256-L269

so I should have actually called this AT_FORALL_QINT_TYPES, but I fix that up in a later PR in this stack.

Reply:
I filed https://github.com/pytorch/pytorch/issues/23440 for addressing that.